src/Proto.Remote/RemoteProcess.cs

```diff
@@ -4,18 +4,22 @@
 //   </copyright>
 // -----------------------------------------------------------------------
 
+using System.Diagnostics;
+
 namespace Proto.Remote
 {
     public class RemoteProcess : Process
     {
         private readonly EndpointManager _endpointManager;
         private readonly PID _pid;
         private PID? _endpoint;
+        private long _lastUsedTick;

```

Comment:
This mixes the usage of Stopwatch.GetTimestamp and TimeSpan ticks. As far as I know, that won't be valid for all cases. Simply storing DateTime instance assigned from DateTime.UtcNow, or using DateTime.UtcNow.Ticks should be fast and reliable.

Reply:
I was hoping to avoid the allocation overhead of that :)

Fixed the cutoff calculation instead.

Thanks for spotting it :)

Reply:
The current solution looks good!

Hmm, I think DateTime.UtcNow doesn't allocate, as the DateTime is a struct. Date literally contains only a LONG member inside, no different than storing Ticks as long. https://source.dot.net/#System.Private.CoreLib/DateTime.cs,143

Reply:
You are correct, it does not allocate :)  It is however a little slower than the stopwatch, since it handles leap seconds etc. We don't need the absolute time here, just how long it was since it was used, so I think the stopwatch ticks is the cheapest way to handle that.